### DISCUSSION TITLE
useQuery / fetching forever

### DISCUSSION THREAD

**Discussion by bobmcallan** [urql-graphql/urql#1711] [Q&A] [answered]
I'm using Vue 3, Composition API following the doco as close as possible.

Randomly the query returns a result, and all status 200, etc, however the result.fetching remains true. 

Is there a timeout function or ? to ensure this is validates.

Thanks,
Bob

**Comment by kitten** [comment, marked as answer]
sorry, I have to convert this to a discussion, since it seems more like a usage issue to me.

that said, we can't really help with this or look into this without a reproduction

**Reply by bobmcallan** [reply]
Yes, 'reproduction' is difficult. All I can say is that is not the server, as all indications (network/response, etc) are correct. The data is returned to the component and loaded, however the fetching var is not set to false. 

i.e. everything has worked, however the query is not closed? I've covered with result.fetching || result.data.

**Comment by eduardodgarciac** [comment]
In my case, is working fine when the query is resolved from network but remains true when the query is resolved from cache.

**Comment by misidoro** [comment]
It's happening the same to me. I am doing multiple queries to the same graphql endpoint (just different variables) at the same time and one of the queries tend to be stuck forever fetching. Is there a timeout for urql to be able to understand that the query was successful? 
Not using any exchanges currently.

I am using react with:
* "@urql/core": "2.3.3",
* "urql": "2.0.5",

I've updated to the latest version and keeps happening:
* "@urql/core": "2.4.1",
 * "urql": "2.1.2",

**Comment by lrstanley** [comment]
See: https://github.com/FormidableLabs/urql/issues/2441 -- there is at least a temporary workaround for the situation I ran into with it fetching not timing out/returning, I am not sure if it's the exact problem others here are having.
